app/assets/javascripts/beak/widgets/ractives/tab.coffee

```diff
@@ -0,0 +1,78 @@
+
+
+RactiveTabWidget = Ractive.extend({
+
+  data: () -> {
+      name:             undefined  # String
+    , title:            undefined  # String
+    , show:             false      # Boolean
+    , 'focus-target':   undefined  # String (Selector)
+    # https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView
+    , 'scroll-target':   undefined # String (Selector) defaults to focus-target
+    , 'scroll-behavior': 'auto'    # String ('auto' | 'smooth')
+    , 'scroll-block':    'nearest' # String ('start' | 'center' | 'end' | 'nearest')
+    , 'scroll-inline':   'nearest' # String ('start' | 'center' | 'end' | 'nearest')
+    , 'scroll-container': 'all'    # String ('all' | 'nearest')
+  }
+
+  focus: ->
+    target = @find(@get('focus-target'))
+    if target?
+      options = @get('scroll-options') or {}
+      target.focus(options)
+    return
+
+  scrollIntoView: ->
+    name = @get('name') or 'unknown'
```

Comment:
There are a lot of places where you use `or` like this.  In CoffeeScript, I prefer to only see `or` used for logical comparisons.  `@get('name') ? 'unknown'` achieves what you want.